### PR TITLE
Fixes the syntax error with contextual component

### LINKED ISSUE #230: SyntaxError for nested block params
```hbs
    {{#x-select value=letter action=(action (mut letter)) as |xs|}}
        <option>Filter by the letter...</option>
        {{#each alphabet as |letter|}}
            {{#xs.option value=letter}}{{ letter }}{{/xs.option}}
        {{/each}}
    {{/x-select}}
```

leads to

```
undefined: `xs.option` is not a valid name for a helper on line 23. (cargo/templates/crates): 
`Error: `xs.option` is not a valid name for a helper on line 23.
    at SyntaxError.Error (native)
    at new SyntaxError (/Users/tbieniek/Code/crates.io/node_modules/@glimmer/syntax/dist/commonjs/es5/lib/errors/syntax-error.js:14:27)
    at assertIsSimplePath (/Users/tbieniek/Code/crates.io/node_modules/ember-template-lint/node_modules/@glimmer/compiler/dist/commonjs/es5/lib/template-compiler.js:476:15)
    at TemplateCompiler.prepareHelper (/Users/tbieniek/Code/crates.io/node_modules/ember-template-lint/node_modules/@glimmer/compiler/dist/commonjs/es5/lib/template-compiler.js:342:9)
    at TemplateCompiler.block (/Users/tbieniek/Code/crates.io/node_modules/ember-template-lint/node_modules/@glimmer/compiler/dist/commonjs/es5/lib/template-compiler.js:196:14)
    at /Users/tbieniek/Code/crates.io/node_modules/ember-template-lint/node_modules/@glimmer/compiler/dist/commonjs/es5/lib/template-compiler.js:71:25
    at Array.forEach (native)
    at TemplateCompiler.process (/Users/tbieniek/Code/crates.io/node_modules/ember-template-lint/node_modules/@glimmer/compiler/dist/commonjs/es5/lib/template-compiler.js:64:17)
    at Function.compile (/Users/tbieniek/Code/crates.io/node_modules/ember-template-lint/node_modules/@glimmer/compiler/dist/commonjs/es5/lib/template-compiler.js:57:32)
    at precompile (/Users/tbieniek/Code/crates.io/node_modules/ember-template-lint/node_modules/@glimmer/compiler/dist/commonjs/es5/lib/compiler.js:45:60)
    at Linter.verify (/Users/tbieniek/Code/crates.io/node_modules/ember-template-lint/lib/index.js:110:7)
    at TemplateLinter.processString (/Users/tbieniek/Code/crates.io/node_modules/ember-cli-template-lint/broccoli-template-linter.js:101:28)
    at Promise.then.result.output (/Users/tbieniek/Code/crates.io/node_modules/broccoli-persistent-filter/lib/strategies/persistent.js:41:23)
    at initializePromise (/Users/tbieniek/Code/crates.io/node_modules/rsvp/dist/rsvp.js:589:5)
    at new Promise$1 (/Users/tbieniek/Code/crates.io/node_modules/rsvp/dist/rsvp.js:1077:33)
    at /Users/tbieniek/Code/crates.io/node_modules/broccoli-persistent-filter/lib/strategies/persistent.js:40:18`@ 1 ms
```

am I doing something wrong?

```
$ yarn list ember-cli-template-lint
yarn list v0.25.1
└─ ember-cli-template-lint@0.7.0
```

### PR DESCRIPTION
**Pull request by karthiick**
Fixes #230

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -45,8 +47,8 @@ class Linter {
     }
 
     let rules = this.config.loadedRules;
+    let astPlugins = [TransformDotComponentInvocation];
```
Comment (karthiick) [inline lib/index.js]: @rwjblue I am not understanding, how the linter errors are shown for the first time when i run `ember s`. Is it using some cached output? It seems like that.

File: lib/plugins/transform-dot-component-invocation.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    visitors: {
+      MustacheStatement: (node) => {
+        if (isInlineInvocation(node.path, node.params, node.hash)) {
+          wrapInComponent(node, b);
+        }
+      },
+      BlockStatement: (node) => {
+        if (isMultipartPath(node.path)) {
+          wrapInComponent(node, b)
```
Comment (karthiick) [inline lib/plugins/transform-dot-component-invocation.js]: missing ;

File: lib/plugins/transform-dot-component-invocation.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        }
+      }
+    }
+  }
+}
```
Comment (karthiick) [inline lib/plugins/transform-dot-component-invocation.js]: ;

File: lib/plugins/transform-dot-component-invocation.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
+
```
Comment (rwjblue) [inline lib/plugins/transform-dot-component-invocation.js]: Can you link to this same file in emberjs/ember.js and explain why we have to include it here? 

Also, can you remove some of these docs (I think a single example is probably enough)?

Comment (karthiick) [inline lib/plugins/transform-dot-component-invocation.js]: @rwjblue sure, will do that.

**Comment by rwjblue** [comment]
This looks good to me.  Can you add a test like [this one](https://github.com/rwjblue/ember-template-lint/blob/master/test/unit/recommended-config-test.js#L36) to ensure that we have some coverage of the original regression?

Something like this should do the trick:

```js
ensureValid(`{{#foo-bar as |baz|}}{{#baz.derp}}{{/baz.derp}}{{/foo-bar}}`);
```

**Comment by karthiick** [comment]
I have added the test case, let me know if it needs any fine tuning. // @rwjblue

File: test/unit/plugins/transform-dot-component-invocation.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    it(`passes with: \`${source}\``, function() {
+      let linter = new Linter({
+        console: mockConsole
+      });
+
+      expect(linter.verify({ source, moduleId: 'some/thing.hbs' })).to.deep.equal([]);
+    });
+  }
+
+  ensureValid('{{#foo-bar as |baz|}}{{#baz.derp}}{{/baz.derp}}{{baz.derp}}{{/foo-bar}}');
```
Comment (rwjblue) [inline test/unit/plugins/transform-dot-component-invocation.js]: Ahh, sorry about that! I was actually meaning that you would add just this line to the `test/unit/recommended-config-test.js` file.

**Comment by karthiick** [comment]
@rwjblue no problem, i have modified it.

**Comment by rwjblue** [comment]
Thank you for your help on this!!!

**Pull request merged**